lib/makeCmdTasks.js

```diff
@@ -27,25 +27,42 @@ module.exports = async function makeCmdTasks({ commands, files, gitDir, shell })
     const resolvedArray = Array.isArray(resolved) ? resolved : [resolved] // Wrap non-array command as array
 
     for (const command of resolvedArray) {
-      let title = isFn ? '[Function]' : command
+      let title = isFn ? '[Function] ' : ''
 
-      if (isFn) {
-        // If the function linter didn't return string | string[]  it won't work
-        // Do the validation here instead of `validateConfig` to skip evaluating the function multiple times
-        if (typeof command !== 'string') {
-          throw new Error(
-            createError(
-              title,
-              'Function task should return a string or an array of strings',
-              resolved
-            )
+      // If the function linter didn't return string | string[]  it won't work
+      // Do the validation here instead of `validateConfig` to skip evaluating the function multiple times
+      if (isFn && typeof command !== 'string') {
+        throw new Error(
+          createError(
+            title.trim(),
+            'Function task should return a string or an array of strings',
+            resolved
           )
+        )
+      }
+
+      const words = command.trim().split(' ') // Tokenize the command into words
+
+      if (isFn) {
+        const indexOfFile = words.findIndex(word => files.includes(word)) // index of first file in the command
```

Comment:
I fear this might not work as-is, since the task function can format files.

It might be necessary to do something like:

```js
const fileBasenames = files.map(file => path.basename(file))
const indexOfFile = words.findIndex(word => fileBasenames.some(basename => word.includes(basename)))
```

to match only the basename (ie. `file.js`).

This is bound to have performance indications, however.

Reply:
Yeah, I also realized tokenizing by splitting on space isn't going to work properly either, if there are spaces inside the filename. Let me revise it a bit more, to try and handle the basename logic above. Is the function formatting files the reason Windows test is failing as well?

Reply:
Can’t say for sure, though I would think the `files` available in this method are normalized the same way; apparently not.